### PR TITLE
Hanlde corner cases for empty or null service name

### PR DESCRIPTION
**Pull request by SylvainJuge**
## What does this PR do?

This PR tests and improve heuristics to get service name from JVM system properties.
Current implementation might allow to have empty values and did not handled properly values with spaces.

## Checklist

- [x] My code follows the [style guidelines of this project](CONTRIBUTING.md#java-language-formatting-guidelines)
- [x] I have rebased my changes on top of the latest master branch
- ~~I have made corresponding changes to the documentation~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#testing) pass locally with my changes
- [x] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
- ~~I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)~~
- ~~Added an API method or config option? Document in which version this will be introduced~~
- ~~Added an instrumentation plugin? How did you make sure that old, non-supported versions are not instrumented by accident?~~

## Author's Checklist
- [x] manual testing with Weblogic server

### CODE REVIEW

File: apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/ServiceNameUtil.java [hunk 1]
```diff
@@ -84,12 +89,14 @@ public static String replaceDisallowedChars(String serviceName) {
     @Nullable
     private static String parseJarCommand(String command) {
         final String[] commandParts = command.split(" ");
+        String result = null;
         for (String commandPart : commandParts) {
             if (commandPart.endsWith(".jar")) {
-                return removeVersionFromJar(removePath(removeJarExtension(commandPart)));
+                result = removeVersionFromJar(removePath(removeJarExtension(commandPart)));
+                break;
```
Comment (SylvainJuge) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/ServiceNameUtil.java]: using a break here allows to maximize coverage, otherwise the `return null;` statement was dead code
